components/private_channel/browser/private_channel_service.cc

```diff
@@ -0,0 +1,320 @@
+      net::LOAD_DO_NOT_SEND_COOKIES | net::LOAD_DO_NOT_SAVE_COOKIES |
+      net::LOAD_BYPASS_CACHE | net::LOAD_DISABLE_CACHE |
+      net::LOAD_DO_NOT_SEND_AUTH_DATA;
+    network::mojom::URLLoaderFactory* loader_factory =
+      g_browser_process->
+        system_network_context_manager()->GetURLLoaderFactory();
+
+    net::NetworkTrafficAnnotationTag traffic_annotation =
+      net::DefineNetworkTrafficAnnotation("brave_private_channel_meta", R"(
+        semantics {
+          sender:
+            "Brave Private Channel Service"
+          description:
+            "Requests Metadata from Private Channel Server"
+            "to setup private two-party computation channel."
+          trigger:
+            "When starting a new attestation based on Private Channels"
+          destination: WEBSITE
+        }
+        policy {
+          cookies_allowed: NO
+          setting:
+            "This feature cannot be disabled by settings."
+          policy_exception_justification:
+            "Not implemented."
+        })");
+
+    http_loader_ = network::SimpleURLLoader::Create(
+      std::move(resource_request), traffic_annotation);
+    http_loader_->SetAllowHttpErrorResults(true);
+
+    http_loader_->DownloadToString(
+      loader_factory,
+      base::BindOnce(&PrivateChannel::OnPrivateChannelMetaLoadComplete,
+        base::Unretained(this)),
+        kMaxPrivateChannelServerResponseSizeBytes);
+  }
+
+  void PrivateChannel::OnPrivateChannelMetaLoadComplete(
+      std::unique_ptr<std::string> response_body) {
+        LOG(INFO) << "PrivateChannel::OnPrivateChannelMetaLoadComplete";
+
+        int response_code = -1;
+        if (http_loader_->ResponseInfo() &&
+            http_loader_->ResponseInfo()->headers)
+          response_code =
+              http_loader_->ResponseInfo()->headers->response_code();
+
+        const std::string safe_response_body =
+          response_body ? *response_body : std::string();
+
+        if (http_loader_->NetError() != net::OK || response_code < 200 ||
+            response_code > 299) {
+          LOG(ERROR) << "Failed to fetch metadata from private channel server"
+                     << ", error: " << http_loader_->NetError()
+                     << ", response code: " << response_code
+                     << ", payload: " << safe_response_body
+                     << ", url: " << http_loader_->GetFinalURL().spec();
+          return;
+        }
+        // We expect the public key response from the server to be correct,
+        // if that's not the case, the protocol will eventually fail gracefully
+        this->FirstRoundProtocol(safe_response_body.c_str());
+  }
+
+  void PrivateChannel::FirstRoundProtocol(const char* server_pk) {
+    LOG(INFO) << "PrivateChannel::FirstRoundProtocol";
+
+    // TODO(gpestana): refactor and extract signals
+    std::string s = "";
+    const char* input[] = {
+      s.c_str(), s.c_str(), s.c_str(), s.c_str(), s.c_str(),
```

Comment:
why are you initializing this with `s.c_str()`?

Reply:
These are mock values and will be removed asap, once I include the correct signal extractor.